### PR TITLE
Use await syntax in analytics

### PR DESCRIPTION
**Pull request by maxjeffos**
- [X] Follows [CONTRIBUTING](https://github.com/snyk/snyk/blob/master/.github/CONTRIBUTING.md) rules

#### What does this PR do?
Switches from callback syntax to await syntax in analytics module.

### CODE REVIEW

File: src/lib/analytics.ts [hunk 1]
```diff
@@ -73,74 +73,70 @@ export function postAnalytics(
...
-        json: true,
-        method: 'post',
-        headers: headers,
-      });
-    })
-    .catch((error) => {
-      debug('analytics', error); // this swallows the analytics error
+  const snykVersion = await version.getVersion();
+
+  try {
```
Comment (JackuB) [inline src/lib/analytics.ts:78]: Why is the getVersion outside of try/catch?

File: src/lib/protect/apply-patch.js [hunk 2]
```diff
@@ -5,7 +5,7 @@ const exec = require('child_process').exec;
 const path = require('path');
 const fs = require('fs');
 const semver = require('semver');
-const errorAnalytics = require('../analytics').postAnalytics;
+const analytics = require('../analytics');
```
Comment (JackuB) [inline src/lib/protect/apply-patch.js:8]: Could we use modules?

```suggestion
import { addDataAndSend }  from '../analytics';
```

File: src/lib/analytics.ts (hunk 1 above)
Comment (maxjeffos) [reply src/lib/analytics.ts]: because it was before, but that's not a very good reason... fixed

File: src/lib/protect/apply-patch.js (hunk 2 above)
Comment (maxjeffos) [reply src/lib/protect/apply-patch.js]: 👍

**Review by JackuB — APPROVED** [review]

**Pull request merged**